config/base.yml

```diff
@@ -29,6 +29,10 @@ cache:
     - $HOME/.composer/cache/files
 
 before_script:
+  # Dynamically work out COMPOSER_ROOT_VERSION based on the 4 / 4.8 branch naming convention
+  # $TRAVIS_BRANCH documentation - https://docs.travis-ci.com/user/environment-variables/#default-environment-variables
+  - if [[ "$TRAVIS_BRANCH" =~ ^([0-9\.]+)$ ]]; then export COMPOSER_ROOT_VERSION="${TRAVIS_BRANCH}.x-dev"; fi
```

Comment:
How will this work with tags? The travis doc says:
> for builds triggered by a tag, this is the same as the name of the tag (TRAVIS_TAG).

Ideally, `TRAVIS_BRANCH` equals something like `1.2.3` or `1.2.3-beta1`, then that's what should be assigned to `COMPOSER_ROOT_VERSION`.

Reply:
I'll update, I think there's 3 scenarios:
- 4 / 4.8 branches - they need `${TRAVIS_BRANCH}.x-dev`
- branches non starting with a number e.g. master - they need `dev-${TRAVIS_BRANCH}` (this is much less common)
- tags (starts with number, 2 dots), they should just be left alone as `$TRAVIS_BRANCH`

Reply:
This seemed like it should work, tested as a local bash script, I'll compress it down to a long one-liner for travis
```
TRAVIS_BRANCH=4.8

if [[ "$TRAVIS_BRANCH" =~ ^[1-9]$ ]] || [[ "$TRAVIS_BRANCH" =~ ^[1-9]\.[0-9]+$ ]]; then
    # numeric branch - 4 / 4.8
    export COMPOSER_ROOT_VERSION="${TRAVIS_BRANCH}.x-dev"
elif [[ "$TRAVIS_BRANCH" =~ ^[1-9]\.[0-9]+\.[0-9]+ ]]; then
    # tag - 1.2.3 / 1.2.3-beta1
    export COMPOSER_ROOT_VERSION="${TRAVIS_BRANCH}"
else
    # non-numeric branch - master / some-branch
    export COMPOSER_ROOT_VERSION="dev-${TRAVIS_BRANCH}"
fi

echo "$COMPOSER_ROOT_VERSION"

```